File: C/snippets/fizzbuzz.c
#include <stdio.h>
#include <string.h>
#include <stdlib.h>

int main(int argc, char *argv[])
{
  int fizzbuzz;
  char result[11] = "";

  if (argc < 2)
  {
    printf("Usage: %s <number>\n", argv[0]);
    return 1;
  }

  fizzbuzz = atoi(argv[1]);

  if (fizzbuzz % 3 == 0)
  {
    strcat(result, "fizz");
  }

  if (fizzbuzz % 5 == 0)
  {
    strcat(result, "buzz");
  }

  if (result[0] == '\0')
  {
    printf("%d\n", fizzbuzz);
    return 0;
  }

  printf("%s\n", result);
  return 0;
}

File: Ocaml/snippets/fizzbuzz.ml
let fizzbuzz n = 
  match n with
  | n when n mod 3 = 0 && n mod 5 = 0 -> "fizzbuzz"
  | n when n mod 3 = 0 -> "fizz"
  | n when n mod 5 = 0 -> "buzz"
  | _ -> string_of_int n 
;;